src/cuda.hpp

```diff
@@ -172,6 +178,7 @@ class GPUWait {
   void signal();
  private:
   int32_t* wait_sync __attribute__((aligned(64)));
```

Comment:
This is not part of your change, but how does the aligned attribute work here? I assume it is to make the address of wait_sync be aligned at 64-byte boundaries, does it make any difference? Also, how this is implemented when the variable is actually a field of a class? Does a compiler fill proper padding when necessary?

Reply:
I haven't benchmarked it in detail, but it's meant to avoid false sharing in cache lines. The compiler will pad the structure/class as needed to enforce the alignment (this can change the size of the struct/class).